### DISCUSSION TITLE
Min/Max and TRP warnings

### DISCUSSION THREAD

**Discussion by DeNohvo** [Jyers/Marlin#51] [Q&A] [answered]
Posting this from reddit thread for awareness. Are these warnings working?

Here is my test methods in stock firmware.

1. Unplugged printer PSU and let it cool down, then powered it by usb and set nozzle temp to 200C. Within 2 minutes I got "Nozzle is too lowperture" as it raised the temps ever so slowly.

2. Powered printer normally and set nozzle to 200C, once it reached 200C I left it to stabilize for a few seconds. Once it was reading 200C I once again unplugged PSU and quickly switched to usb. After it powered on with usb, temp was at about 187C I again set the nozzle to 200C. The temp continued to drop and within 2 mins I got "Nozzle is too lowperture".

3. Powered on printer normally and set nozzle to 200C, once it reached 200C i left it alone to stabilize. I then plugged in the usb charger and turned the switch off on the PSU. The board stayed powered on but the heater, fan turned off and temps started to drop, within 2 minutes i got "WARNINGS THERMAL RUNAWAY"

**Comment by t3chguy** [comment]
The board/Marlin likely has no way to know the 24V power is on.
So when it is trying to raise temperature (opens the low side MOSFET) and the temperature doesn't go up as expected, that'll cause a thermal runaway. 

In any case, that's entirely unrelated to this fork which is rewriting the LCD library for the Creality DWIN screen.

Min/Max and TRP are untouched.

**Reply by Jyers** [reply]
I believe he is referring to the warning that popup during these event rather than the actual TPR itself. Which I will be adding in the next release.

**Reply by DeNohvo** [reply]
So am i correct to assume that TRP is in fact enabled in marlin but there is just no output on the screen to notify the user?

Meaning in its current state if TRP happens during a real print the printer and heaters will shut off but the screen will still think everything is fine?

**Reply by Benetos** [reply]
Thats my understaing of the main marlin branch. Either it comes enbabled or if you enable it there is no warning or popup when it triggers. It simply won't heat up the bed or it shuts off the elements.

**Reply by DeNohvo** [reply]
I tried build 1.1 and do get the TRP warning but unable to generate the Nozzle is too lowperture warnings from steps 1 and 2

**Reply by Jyers** [reply]
As far as I can tell the "nozzle is too low temperature" warning is something creality added on top of base marlin. There is not call from marlin for such a popup, so this firmware will not show one.

**Reply by DeNohvo** [reply]
So isn't the Min Temp warning from marlin? That's what I assumed it to be, with the way I am able to get it to trigger when using a usb charger.

Others that got this warning from reddit had bad thermistors so it does make sense to me that this is min temp warning. I haven't found any other threads for the V2 that actually give a min temp warning, if your able to associate  "Nozzle is too lowperture" with the marlin min temp warning I can test it out to confirm in the next release.

https://www.reddit.com/r/ender3v2/comments/jh4uda/has_anyone_come_across_this_before_was_printing/
https://www.reddit.com/r/ender3v2/comments/kjfjcr/i_just_bought_the_ender_3_v2_have_never_owned_a/

**Reply by DeNohvo** [reply]
Here is a thread for "Nozzle is too higherature" 

https://www.reddit.com/r/ender3v2/comments/jq8ktf/ender_3_v2_nozzle_is_too_higherature_error/

So again i think these are the min/max warnings just with poor english but should be associated with the min/max warnings from marlin much the same way you did with TRP.

**Reply by Benetos** [reply]
My only question is have you done your own PID Auto tune sequence? If not then it's possible the default PID values are to much and it's causing the issue. 

I would suggest doing that abs plugging in your own values and build out the firmware your self and see if it solves the warnings.

**Reply by DeNohvo** [reply]
I'm not actually getting these warnings on my printer during normal prints, the printer itself is fine and runs without issue.

I'm forcing these warnings to make sure the saftey checks and warnings are in place and warnings are displayed correctly if/when an real issue happens.

I don't want to be in a situation that by installing CFW I'm removing safety features or even if they are in place there is no visual notification.

The printer should hault and then notify the user if there is a min/max or trp error in marlin.

**Comment by Jyers** [comment, marked as answer]
Alright, I have extensively checked the code and it would seem there is no base marlin equivalent to the way creality was handling these popups. They had added in code that displayed temp to high/low popups no matter what when a temp error was detected, while base marlin only displays a popup if the issue causes a printer kill event. This may mean that the my firmware will not display a popup when creality's may have, but I can assure you my firmware is 100% up to the same safety standard as any base marlin build and will display important user info such as kill events and errors when such a popup is called for.

**Reply by DeNohvo** [reply]
Understood
